tests/integration/nvidia_setup/test_prune_foreign_kernels.py

```
@@ -0,0 +1,160 @@
+    assert_exit_code,
+    assert_output_contains,
+    assert_output_not_contains,
+)
+from tests.helpers.docker_test import DockerTestRunner
+
+# Test script lives with tests and is copied into the package at run time.
+_PRUNE_SCRIPT_SOURCE = Path(__file__).parent / "run_prune_foreign_kernels_test.sh"
+_PRUNE_SCRIPT_DEST = "skyhook_dir/steps/run_prune_foreign_kernels_test.sh"
+
+
+def _run_prune(runner, running, installed_packages, keep_extra=""):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_run_prune`

(ANN202)